sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -413,9 +413,13 @@ case class DataSource(
         } else {
           val globbedPaths = checkAndGlobPathIfNecessary(
             checkEmptyGlobPath = true, checkFilesExist = checkFilesExist)
-          val index = createInMemoryFileIndex(globbedPaths)
+          val fileStatusCache = FileStatusCache.getOrCreate(sparkSession)
+          val indexInSchemaInferring = new InMemoryFileIndex(
+            sparkSession, globbedPaths, options, userSpecifiedSchema, fileStatusCache)
```

Comment:
What's diff from calling `createInMemoryFileIndex` here?

Reply:
`fileStatusCache`. I want to re-use it.